conanfile.py

```diff
@@ -52,10 +56,15 @@ def source(self):
         self.run("gclient config --spec=\"%s\"" % self._make_spec(), run_environment=True)
         self.run("gclient sync --no-history", run_environment=True)
 
-        patch_base = os.path.join(self._source_dir, "third_party/mini_chromium/mini_chromium");
+        patch_base = os.path.join(
+            self._source_dir, "third_party/mini_chromium/mini_chromium")
         tools.patch(base_path=patch_base,
                     patch_file="patches/buildsystem-adaptions.patch")
 
+        if self.options.force_embedded_zlib:
+            tools.patch(base_path=patch_base,
+                        patch_file="patches/force-embedded-zlib.patch")
+
```

Comment:
this does not work in the general case, because conan reuses the same sources, regardless of the options used. In short the options values must never be used in `def source(self):`. This patching should occur in `def build(self)` instead